core/src/main/java/org/apache/spark/shuffle/sort/ShuffleExternalSorter.java

```diff
@@ -254,7 +252,11 @@ public long spill(long size, MemoryConsumer trigger) throws IOException {
       spills.size() > 1 ? " times" : " time");
 
     writeSortedFile(false);
-    final long spillSize = freeMemory();
+    long spillSize = freeMemory();
```

Comment:
I'm sorry that I misread this last night, this is spillSize (the number of bytes written into disk), not the amount of freed memory, so we don't need to add the amount from inMemSorter.

Sorry again.